core-common/src/main/java/org/zalando/nakadi/service/subscription/zk/ZkSubscriptionClient.java

```diff
@@ -202,21 +203,18 @@ public Partition[] getPartitions() {
         }
 
         public Topology withUpdatedPartitions(final Partition[] partitions) {
-            final Partition[] resultPartitions = Arrays.copyOf(this.partitions, this.partitions.length);
+            final var resultPartitions = new ArrayList<>(Arrays.asList(this.partitions));
             for (final Partition newValue : partitions) {
-                int selectedIdx = -1;
-                for (int idx = 0; idx < resultPartitions.length; ++idx) {
-                    if (resultPartitions[idx].getKey().equals(newValue.getKey())) {
-                        selectedIdx = idx;
-                    }
-                }
-                if (selectedIdx < 0) {
-                    throw new NakadiBaseException(
-                            "Failed to find partition " + newValue.getKey() + " in " + this);
+                final var selected = IntStream.range(0, resultPartitions.size())
+                        .filter(idx -> resultPartitions.get(idx).getKey().equals(newValue.getKey()))
+                        .findFirst();
+                if (selected.isPresent()) {
+                    resultPartitions.set(selected.getAsInt(), newValue);
+                } else {
+                    resultPartitions.add(newValue);
                 }
-                resultPartitions[selectedIdx] = newValue;
             }
-            return new Topology(resultPartitions, Optional.ofNullable(version).orElse(0) + 1);
+            return new Topology(resultPartitions.toArray(new Partition[0]), Optional.ofNullable(version).orElse(0) + 1);
```

Comment:
```suggestion
            return new Topology(resultPartitions.toArray(Partition[]::new), Optional.ofNullable(version).orElse(0) + 1);
```

Reply:
Such method call would effectively delegate to `.toArray(new Partition[0])`.
What would be the benefit of such indirection?